site/en/blog/service-worker-static-routing-api-origin-trial/index.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: 'layouts/blog-post.njk'
+title: Service Worker Static Routing API Origin Trial
+description: |
+  Learn about the first phase of the Service Worker Static Routing—now in an Origin Trial—a new API for declaratively specifying how routes should be handled (or not) by your Service Worker.
+authors:
+  - brendankenny
+date: 2023-11-10
+hero: 'image/MtjnObpuceYe3ijODN3a79WrxLU2/r0jFyrDEH1gy1hCYnVb1.jpg'
+alt: >
+  A stock photo of a card catalog.
+tags:
+  - service-worker
+  - origin-trials
+  - javascript
+  - performance
+---
+
+Service workers are a powerful tool for allowing websites to work offline and create specialized caching rules for themselves. A service worker `fetch` handler sees every request from a page it controls, and can decide if it wants to serve a response to it from the service-worker-cache files, or even rewrite the URL to fetch a different response entirely—for instance, based on local user preferences.
+
+However, there can be a performance cost to service workers when a page is loaded for the first time in a while and the controlling service worker isn't currently running. Since all fetches need to happen through the service worker, the browser has to wait for the service worker to start up and run to know what content to load. This startup cost can be small, but significant, for developers using service workers to improve performance through caching strategies.
+
+[Navigation preload](https://web.dev/blog/navigation-preload) is one approach to solving the problem—allowing navigation requests to be made over the network in parallel to service-worker startup—but it's limited to initial navigation requests and still includes the service worker in the critical path. Since navigation preload launched, there have been multiple efforts to develop a more general solution to the problem space, including ways for some requests to not be blocked on service worker startup at all.
```

Comment:
```suggestion
[Navigation preload](https://web.dev/blog/navigation-preload) is one approach to solving the problem—allowing navigation requests to be made over the network in parallel to service worker startup—but it's limited to initial navigation requests and still includes the service worker in the critical path. Since navigation preload launched, there have been multiple efforts to develop a more general solution to the problem space, including ways for some requests to not be blocked on service worker startup at all.
```